metals/src/main/scala/scala/meta/internal/metals/WorkspaceSymbolProvider.scala

```diff
@@ -58,6 +62,16 @@ final class WorkspaceSymbolProvider(
     inDependencies.search(query, visitor)
   }
 
+  def searchMethods(
+      query: String,
+      visitor: SymbolSearchVisitor,
+      target: Option[BuildTargetIdentifier],
+  ): SymbolSearch.Result = {
+    workspaceMethodSearch(query, visitor, target)
+    if (query.isEmpty()) SymbolSearch.Result.INCOMPLETE
```

Comment:
if it's `SymbolSearch.Result.COMPLETE` when `query.isEmpty`, once the client requests completion on `foo.` with an empty query, the client won't request completions when we type `foo.x`

as LSP specification says

```
	/**
	 * This list is not complete. Further typing should result in recomputing
	 * this list.
	 *
	 * Recomputed lists have all their items replaced (not appended) in the
	 * incomplete completion sessions.
	 */
	isIncomplete: boolean;
```
https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_completion